R/db-helpers.R

```diff
@@ -167,43 +167,56 @@ repair_table_names_for_db <- function(table_names, temporary, con) {
   quote_ids(names, con)
 }
 
-get_src_tbl_names <- function(src, schema = NULL) {
+get_src_tbl_names <- function(src, schema = NULL, dbname = NULL) {
   con <- src$con
 
-  if (is_null(schema)) {
-    if (!is_mssql(src) && !is_postgres(src)) {
-      # `src_tbls()` returns system tables and tables in other schemas than default schema only for MSSQL
+  if (!is_mssql(src)) {
+    if (!is_null(dbname)) {
+      warn("Argument `dbname` ignored: currently only supported for MSSQL")
+      # for Postgres:
+      dbname <- NULL
+    }
+    if (!is_postgres(src)) {
+      # any DBMS different from MSSQL or Postgres
+      if (!is_null(schema)) {
+        warn("Argument `schema` ignored: currently only supported for MSSQL and Postgres")
+      }
+      # in any case return `src_tbls(src)` if not MSSQL or Postgres
       return(src_tbls(src))
```

Comment:
Alright, I can do that, it won't change the behavior. 
Only thing is, that then we need to have the same warning about a non-NULL parameter `dbname` twice in the code: once for `if (!is_mssql(src) && !is_postgres(src))` and once for `if (!is_postgres(src))`. I wanted to avoid that.

Reply:
We can extract a function that issues a warning if the argument is non-`NULL` to avoid repetition:

```r
warn_if_non_null <- function(arg) {
  arg_expr <- enexpr(arg)
  if (is.null(arg)) return()
  ...
}
```